store/v2/transient/store.go

```diff
@@ -0,0 +1,44 @@
+package transient
+
+import (
+	dbm "github.com/cosmos/cosmos-sdk/db"
+	"github.com/cosmos/cosmos-sdk/store/types"
+	"github.com/cosmos/cosmos-sdk/store/v2/dbadapter"
+)
+
+var (
+	_ types.KVStore   = (*Store)(nil)
+	_ types.Committer = (*Store)(nil)
+)
+
+// Store is a wrapper for a memory store which does not persist data.
+type Store struct {
+	dbadapter.Store
+	conn dbm.DBConnection
+}
+
+// NewStore constructs a new transient store.
+func NewStore(db dbm.DBConnection) *Store {
+	return &Store{
+		Store: dbadapter.Store{DB: db.ReadWriter()},
+		conn:  db,
+	}
+}
+
+// Implements Store.
+func (ts *Store) GetStoreType() types.StoreType {
+	return types.StoreTypeTransient
+}
+
+// Implements CommitStore
+// Commit cleans up Store.
+func (ts *Store) Commit() (id types.CommitID) {
+	ts.DB.Discard()
+	ts.Store = dbadapter.Store{DB: ts.conn.ReadWriter()}
+	return
+}
+
+func (ts *Store) SetPruning(types.PruningOptions)  {}
+func (ts *Store) GetPruning() types.PruningOptions { return types.PruningOptions{} }
+
+func (ts *Store) LastCommitID() (id types.CommitID) { return }
```

Comment:
Maybe there is a better way to manage this: instead of copying files we can import them from common package? May require some refactoring.

Reply:
There's a way to do it by adapting the tmdb.DB to a DBConnection or vice versa. But, I don't think it buys us much, since this code is very simple. Also if in the future we plan to deprecate/remove tmdb, we would rather have this be independent.